### PR TITLE
Add wrapper class for `GetTxResponse`

### PR DESCRIPTION
**Pull request by keefertaylor**
## High Level Overview of Change

Implement a wrapper class for `GetTxResponse` which conforms to the `RawTransactionStatus` interface.

### Context of Change

rippled implemented protocol buffer support natively, which uses a new and incompatible set of protocol buffers. The implementation in rippled was completed in: https://github.com/ripple/rippled/pull/3159.

`XpringClient` uses a decorator pattern where each implementation must conform to an interface. The interface required for the `getRawTransactionStatus` method requires it return a `RawTransactionStatus`. 

The new protocol buffers from rippled return a `GetTxResponse` protocol buffer which does not conform to this interface. This wrapper class conforms a`GetTxResponse` to this interface. This unblocks implementation of the `getRawTransactionStatus` method in rippled. 

*Status*

- [x] [Generate new versions of the protocol buffers](https://github.com/xpring-eng/Xpring-JS/pull/108)
- [x] [Migrate existing code to a `legacy` subdirectory and namespace](https://github.com/xpring-eng/Xpring-JS/pull/109)
- [ ] Create code that works with the new protocol buffers from the rippled team
   - [x] [Create a new `NetworkClient` with fakes](https://github.com/xpring-eng/Xpring-JS/pull/111)
   - [x] [Create a new `DefaultXpringClient` with tests](https://github.com/xpring-eng/Xpring-JS/pull/114/files)
   - [ ] Read only calls for new `DefaultXpringClient` (In progress)
   - [ ] Write calls for `DefaultXpringClient`
- [ ] Add a boolean parameter to switch between implementations

### Type of Change

- [x] New feature (non-breaking change which adds functionality)

## Before / After

No change. This new class is not exposed to the world or hooked up yet. 

## Test Plan

CI Continues to work.

### CODE REVIEW

File: src/xpring-client.ts [hunk 1]
```diff
@@ -17,12 +16,11 @@ class XpringClient {
    * The XpringClient will use gRPC to communicate with the given endpoint.
    *
    * @param grpcURL The URL of the gRPC instance to connect to.
-   * @param useNewProtocolBuffers If `true`, then the new protocol buffer implementation from rippled will be used. Defaults to false.
    */
-  public constructor(grpcURL: string, useNewProtocolBuffers = false) {
```
Comment (hbergren) [inline src/xpring-client.ts]: What happened to `useNewProtocolBuffers`?

Comment (keefertaylor) [reply src/xpring-client.ts]: Just a bad merge. Sorry for the many balls in the air.

File: src/default-xpring-client.ts [hunk 2]
```diff
@@ -27,6 +28,40 @@ export class XpringClientErrorMessages {
   /* eslint-enable @typescript-eslint/indent */
 }
 
+/**
+ * A private wrapper class which conforms `GetTxResponse` to the `RawTransaction` interface.
+ */
+class GetTxResponseWrapper implements RawTransactionStatus {
```
Comment (hbergren) [inline src/default-xpring-client.ts]: Where is this class used?

**Comment by keefertaylor** [comment]
All good things in time. This class gets hooked up in https://github.com/xpring-eng/Xpring-JS/pull/131.

**Pull request merged**